excavator/utils.py

```diff
@@ -64,8 +64,13 @@ def env_list(name, separator=',', required=False, default=empty):
     done by splitting the string value from the environment on a given
     separator.
     """
-    value = get_env_value(name, required=required, default=default)
+    if required and default is not empty:
+        raise ValueError("Using `default` with `required=True` is invalid")
+
+    value = get_env_value(name, required=required, default=empty)
     if value is empty:
+        if default is not empty:
+            return default
         return []
     # wrapped in list to force evaluation in python 3
     return list(filter(bool, [v.strip() for v in value.split(separator)]))
```

Comment:
Additional here: list-comprehension is generally a style I prefer over `filter`, as it is more explicit about what you get back, and consistent across py2/py3.